src/searchReplace.tsx

```diff
@@ -29,14 +37,52 @@ export class SearchReplaceModel extends VDomModel {
     }
   }
 
+  get caseSensitive(): boolean {
+    return this._caseSensitive;
+  }
+
+  set caseSensitive(v: boolean) {
+    this._caseSensitive = v;
+    this.stateChanged.emit();
+  }
+
+  get wholeWord(): boolean {
+    return this._wholeWord;
+  }
+
+  set wholeWord(v: boolean) {
+    this._wholeWord = v;
+    this.stateChanged.emit();
+  }
+
+  get useRegex(): boolean {
+    return this._useRegex;
+  }
+
+  set useRegex(v: boolean) {
+    this._useRegex = v;
+    this.stateChanged.emit();
+  }
```

Comment:
I would add the new search trigger in the setters. So the model will be consistent and in the view you don't need to call the search update.

> You should set `getSearchString` as private and never trigger it. But always invoke it through the debouncer.

Reply:
done